workflow/installertest.groovy

```diff
@@ -92,41 +205,79 @@ def runShellTest(String imageName, def shellCommands, def stepNames=null) {
           // Second, the sh workflow step often will use the default posix shell
           // The default posix shell does not support pipefail, so we have to invoke bash to get it
           
-          String argument = 'bash -c \'set -o pipefail; '+cmd+" | tee \"testresults/$fileName-$name"+'.log'+'" \''
+          String argument = "${cmd} | tee \"testresults/${fileName}-${name}.log\""
           sh argument
         }
       } catch (Exception ex) {
+        // And archive the test results once we're done.
+        step([$class: 'JUnitResultArchiver', keepLongStdio: true, testResults: "results/*.xml"])
```

Comment:
I inherited this chunk of logic from what came before - I think @svanoort had hit some weird problems with `finally` vs `catch`, but I'm good either way.